### PR TITLE
Remove to hash deprecation

### PR DESCRIPTION
**Pull request by dncrht**
Treat ActionController::Parameters as plan hashes, and keep using plain hashes as such.

@andrewgristwood @dragosmiron @mineshdattani @sivanpatel

### CODE REVIEW

**Comment by dncrht** [comment]
https://trello.com/c/0ULx2WyK/948-clean-up-chopin-deprecation-warnings

File: lib/jimmy/log_entry.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module Jimmy
   class Entry < Hash
     def <<(hash)
-      self.merge!(hash)
+      self.merge!(hash.try(:to_unsafe_h) || hash)
```
Comment (andrewgristwood) [inline lib/jimmy/log_entry.rb]: Why does this need to be wrapped in a try?

Comment (dragosmiron) [inline lib/jimmy/log_entry.rb]: what's the deprecation error for this ?

Comment (dncrht) [inline lib/jimmy/log_entry.rb]: The deprecation comes from the fact people tend to log `params` straight away. Params is no longer a hash, hence it has to be converted to hash @dragosmiron 

But people is sending hashes here too, and they don't provide the to_unsafe_h method, in those cases the original hash is allowed through @andrewgristwood . It's the same technique we already use in chopin https://github.com/simplybusiness/chopin/blob/master/lib/ci5/questionnaire_support.rb#L79

Comment (dragosmiron) [inline lib/jimmy/log_entry.rb]: i see so that hash is not really a hash. can't we just convert to hash before sending here?

Comment (dncrht) [inline lib/jimmy/log_entry.rb]: This is an external library and it by definition, it should not trust the data it receives.
We'd better be defensive, we may not have control over all those places that send a hash or not.

**Comment by dragosmiron** [comment]
>This is an external library and it by definition, it should not trust the data it receives.

doesnt this mean it has to guard against data it does not trust ?
if youre calling any API with some s**t, is not gonna try to convert your data
is just gonna blow

File: lib/jimmy/log_entry.rb (hunk 1 above)
Comment (dragosmiron) [inline lib/jimmy/log_entry.rb]: >This is an external library and it by definition, it should not trust the data it receives.

doesnt this mean it has to guard against data it does not trust ?
if youre calling any API with some s**t, is not gonna try to convert your data
is just gonna blow

Comment (dncrht) [inline lib/jimmy/log_entry.rb]: Closed as we do have control over the places `params` are sent here: only one place in chopin.

**Pull request closed without merging**